File: fix_specific_jsx_errors.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_specific_jsx_errors(file_path):
    """Fix specific JSX errors in a file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix malformed paragraph tags
        content = re.sub(r'<p[^>]*>([^<]+)</p>\s*</p>', r'<p>\1</p>', content)
        
        # Fix missing closing tags for Link components
        content = re.sub(r'<Link[^>]*>([^<]+)\n\s*<ArrowRight[^>]*/>\s*\n\s*</Link>', r'<Link to="/contact" className="inline-flex items-center px-6 py-3 bg-blue-600 text-white font-semibold rounded-lg hover:bg-blue-700 transition-colors">\1\n            <ArrowRight className="ml-2 h-5 w-5" />\n          </Link>', content, flags=re.MULTILINE)
        
        # Fix malformed JSX structure
        lines = content.split('\n')
        fixed_lines = []
        i = 0
        while i < len(lines):
            line = lines[i]
            
            # Fix lines that have malformed closing tags
            if re.search(r'</p>\s*$', line) and i + 1 < len(lines):
                next_line = lines[i + 1]
                if next_line.strip() == '</p>':
                    # Remove duplicate closing tag
                    fixed_lines.append(line)
                    i += 2
                    continue
            
            # Fix lines that have malformed Link components
            if re.search(r'<Link[^>]*>[^<]+$', line) and i + 1 < len(lines):
                next_line = lines[i + 1]
                if re.search(r'<ArrowRight[^>]*/>', next_line):
                    # This is a malformed Link component
                    fixed_lines.append(line + '\n            <ArrowRight className="ml-2 h-5 w-5" />\n          </Link>')
                    i += 2
                    continue
            
            fixed_lines.append(line)
            i += 1
        
        content = '\n'.join(fixed_lines)
        
        # If content was modified, write it back
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed: {file_path}")
            return True
        else:
            print(f"No changes needed: {file_path}")
            return False
            
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    """Fix specific JSX errors in all TSX files"""
    app_dir = "/workspace/app"
    tsx_files = glob.glob(f"{app_dir}/**/*.tsx", recursive=True)
    
    fixed_count = 0
    total_count = len(tsx_files)
    
    print(f"Found {total_count} TSX files to check for specific JSX errors...")
    
    for file_path in tsx_files:
        if fix_specific_jsx_errors(file_path):
            fixed_count += 1
    
    print(f"\nFixed {fixed_count} out of {total_count} files")

if __name__ == "__main__":
    main()

File: fix-specific-jsx-errors.js
const fs = require('fs');
const path = require('path');

// Function to fix specific JSX structure issues
function fixSpecificJSXErrors(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    // Fix the specific pattern: }}\n  );\n}\n\nexport default
    content = content.replace(/}\s*}\s*\n\s*\);\s*\n\s*}\s*\n\s*\n\s*export default/g, '  );\n};\n\nexport default');
    
    // Fix malformed closing tags like </a></div>
    content = content.replace(/<\/a><\/div>/g, '</a>\n          </div>');
    
    // Fix missing closing parentheses for return statements
    if (content.includes('return (') && !content.includes(');')) {
      content = content.replace(/(<\/div>\s*}\s*}\s*)/g, '$1\n  );\n};');
      modified = true;
    }
    
    // Fix extra closing braces
    content = content.replace(/}\s*}\s*$/gm, '  );\n};');
    
    if (content !== fs.readFileSync(filePath, 'utf8')) {
      fs.writeFileSync(filePath, content);
      console.log(`Fixed specific JSX errors in: ${filePath}`);
      modified = true;
    }
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
  }
}

// Function to recursively find and fix all TypeScript/TSX files
function fixAllFiles(dir) {
  const files = fs.readdirSync(dir);
  
  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory() && !file.startsWith('.') && file !== 'node_modules') {
      fixAllFiles(filePath);
    } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
      fixSpecificJSXErrors(filePath);
    }
  });
}

// Start fixing from the workspace root
console.log('Starting to fix specific JSX errors...');
fixAllFiles('/workspace');
console.log('Finished fixing specific JSX errors.');